doc/design/container_logging.md

```diff
@@ -0,0 +1,24 @@
+# Container VM output logging in VIC
+
+When a container VM process produces output, that output is sent down through a serial connection into the vSphere backend where it is written to a logfile. When a user wants to see this output, they use the `docker logs` command, which reads from this logfile and displays the log entries on the command line in order from oldest to newest.
+
+However, in order for users to be able to use the `--since` option with `docker logs`, we must also couple timestamps to these log entries so that the user can selectively filter log messages to be displayed based on when those entries occurred.
+
+### Requirements
+
+The container VM logging mechanism must:
+  1. Wrap each log entry in a JSON struct containing both the log message(s) in that entry, and the timestamp in which that entry occurred. 
```

Comment:
@hickeng thanks for this. I apologize for missing it (I buried it in a rebase before I saw it). Unfortunately, none of these requirements are actually in the issue so I never even considered most of them. I will open a new issue for tagging each entry with the stream it came from. As far as these other design questions are concerned, I think they can be discussed and decided upon independently from stream tagging, timestamping, and `--timestamps` & `--since` implementations.